include/cantera/oneD/Sim1D.h

```diff
@@ -133,8 +133,22 @@ class Sim1D : public OneDim
      */
     void setRefineCriteria(int dom = -1, doublereal ratio = 10.0,
                            doublereal slope = 0.8, doublereal curve = 0.8, doublereal prune = -0.1);
+
+    /**
+     * Set the maximum number of grid points in the domain. If dom >= 0,
+     * then the settings apply only to the specified domain. If dom < 0,
+     * the settings are applied to each domain.  @see Refiner::setMaxPoints.
+     */
     void setMaxGridPoints(int dom = -1, int npoints = 300);
 
+    /**
+     * Get the maximum number of grid points in this domain. If dom >= 0,
+     * then the returned vector has only a single value. If dom < 0,
+     * the vector has as many elements as there are domains.
+     * @see Refiner::getMaxPoints
+     */
+    std::vector<size_t> getMaxGridPoints(int dom = -1);
```

Comment:
I think it would be simpler if this function either (a) always required a domain index and returned just a `size_t` or (b) took no argument and always returned the full vector.